adrs/0002-monorepo.md

```diff
@@ -0,0 +1,34 @@
+# [Transition Stacks to a monorepo]
+
+**Date:** 2022-11-28
+
+**Related PR:** #1191
+
+## Issue
+
+The Stacks repo is growing in complexity and will soon require many different dependencies that are optional for most consumers.
+
+## Decision
+
+Despite shipping many different concepts (atomic classes, CSS components, JS components), the interdependency chains are relatively linear. This means that we can fairly trivially split the repo into smaller portions.
+
+Our current codebase could be split like so:
+- Atomic classes (no dependencies)
+- CSS Components (depends on Atomic components)
+- JS Components (depends on Stimulus, Popper.js, CSS + Atomic components)
+- Documentation site (depends on JS components, 11ty + plugins)
+
+When the codebase expands in the future, we'll be able to add the following without worrying about consumers needing to import dependencies they don't need:
+- Front-end component library or libraries (e.g. React, Web Components)
+- Large, complex components (e.g. autocomplete)
+- Extra build chain items, such as a [Style Dictionary](https://amzn.github.io/style-dictionary/)
```

Comment:
Nice that you mention a design tokens tool here - it might be something worth exploring in the future (especially if we would have to start supporting non-web platforms too). 🙂

Reply:
I did some exploration on this during my personal "Learn, Share, Grow" day. _Technically_, we already support a "non-web" (non-CSS?) platform - Figma 😉